src/System.Management.Automation/utils/tracing/SysLogProvider.cs

```diff
@@ -0,0 +1,504 @@
+            NativeMethods.SysLogPriority.Info
+        };
+
+        /// <summary>
+        /// Logs a activity transfer
+        /// </summary>
+        /// <param name="parentActivityId">The </param>
+        public void LogTransfer(Guid parentActivityId)
+        {
+            // NOTE: always log
+            int threadId = Thread.CurrentThread.ManagedThreadId;
+            string message = string.Format
+            (
+                CultureInfo.InvariantCulture,
+                "({0:X}:{1:X}:{2:X}) [Transfer]:{3} {4}", _commitId, threadId, PSChannel.Operational,
+                parentActivityId.ToString("B"),
+                _activity.ToString("B")
+            );
+
+            NativeMethods.SysLog(NativeMethods.SysLogPriority.Info, message);
+        }
+
+        /// <summary>
+        /// Logs the activity identifier for the current thread.
+        /// </summary>
+        /// <param name="activity">The Guid activity identifier</param>
+        public void SetActivity(Guid activity)
+        {
+            int threadId = Thread.CurrentThread.ManagedThreadId;
+            _activity = activity;
+
+            // NOTE: always log
+            string message = string.Format(CultureInfo.InvariantCulture, "({0:X}:{1:X}:{2:X}) [Activity] {3}", _commitId, threadId, PSChannel.Operational, activity.ToString("B"));
+            NativeMethods.SysLog(NativeMethods.SysLogPriority.Info, message);
+        }
+
+        /// <summary>
+        /// Logs a message that provides a correlation between the full git commit id string and a hash code used for subsequent log entries.
+        /// </summary>
+        static void LogCommit()
+        {
+            int threadId = Thread.CurrentThread.ManagedThreadId;
+            // NOTE: always log
+            string message = string.Format(CultureInfo.InvariantCulture, "({0:X}:{1:X}:{2:X}) [GitCommitId] {3} Hash: {0:X}", _commitId, threadId, PSChannel.Operational, _gitCommitId);
+            NativeMethods.SysLog(NativeMethods.SysLogPriority.Info, message);
+        }
+
+        /// <summary>
+        /// Writes a log entry
+        /// </summary>
+        /// <param name="eventId">The event id of the log entry.</param>
+        /// <param name="channel">The channel to log</param>
+        /// <param name="task">The task for the log entry</param>
+        /// <param name="opcode">The operation for the log entry</param>
+        /// <param name="level">The logging level.</param>
+        /// <param name="keyword">The keyword(s) for the event.</param>
+        /// <param name="args">The payload for the log message</param>
+        public void Log(PSEventId eventId, PSChannel channel, PSTask task, PSOpcode opcode, PSLevel level, PSKeyword keyword, params object[] args)
+        {
+            if (ShouldLog(level, keyword, channel))
+            {
+                int threadId = Thread.CurrentThread.ManagedThreadId;
+
+                StringBuilder sb = new StringBuilder();
```

Comment:
We could reuse a single instance of `StringBuilder` to avoid extra allocations.

Reply:
Fixed